microcosm_eventsource/immutable_event_factory.py

```diff
@@ -0,0 +1,81 @@
+"""
+Factory for immutable events.
+
+In extension of EventFactory, this class provides a native support of event updating container objects based
+on event types
```

Comment:
I think reiterating the naming somewhere (for example this doc string) would be good. E.g. something like:


```
Given an event stream associated with a `container`, e.g. `ProjectEvent` and 'Project`, a `mutator` takes a container object and an event and makes a change on the `container` object.
```